### PR TITLE
Allow 'same' padding

### PR DESCRIPTION
**Pull request by qlemaire22**
I think that it could be interesting to allow the 'causal' and the 'same' padding. The 'same' padding creates non-causal dilated convolution (see figure below) and it gives the possibility to take into account information from the future when the use case allows it.

I use it in my project and it gave me some better results for applications where I could use information from the future.

Tell me if you think if you agree and if I should make further changes in order for it to be merged.

tcn-nc

Edit: Correct figure

### CODE REVIEW

**Comment by philipperemy** [comment]
@qlemaire22 looks great! Let's get it merged :)

**Comment by philipperemy** [comment]
Let me know when you think it's finalized. I can have a close look at that time. Also, should you need help, please do let me know

**Comment by qlemaire22** [comment]
I have added it to the main doc with the picture I put above. I think that it is all ok.
Please don't hesitate to modify things if you disagree and to tell me if I need to do further changes.

File: tcn/tcn.py [hunk 1]
```diff
@@ -134,11 +137,16 @@ def __init__(self,
...
         self.nb_filters = nb_filters
+        self.padding = padding
 
         # backwards incompatibility warning.
         # o = tcn.TCN(i, return_sequences=False) =>
         # o = tcn.TCN(return_sequences=False)(i)
 
+        if padding != 'causal' and padding != 'same':
+            print("Only 'causal' or 'same' paddings are compatible for this layer.")
+            raise Exception()
```
Comment (philipperemy) [inline tcn/tcn.py]: raise ValueError(error_msg) instead.

File: README.md [hunk 2]
```diff
@@ -100,6 +100,7 @@ model.fit(x, y) # Keras model.
 - `dilations`: List. A dilation list. Example is: [1, 2, 4, 8, 16, 32, 64].
 - `nb_stacks`: Integer. The number of stacks of residual blocks to use.
 - `activation`: String. The activations to use in the residual blocks (norm_relu, wavenet, relu...).
+- `padding`: String. The padding to use in the convolutions. 'causal' for a causal network (as in the original implementation) and 'same' for a non-causal network.
```
Comment (philipperemy) [inline README.md]: In the signature, add padding='causal'

tcn.TCN(nb_filters=64, kernel_size=2, nb_stacks=1, dilations=None, activation='norm_relu', use_skip_connections=True, dropout_rate=0.0, return_sequences=True, name='tcn')

To

tcn.TCN(nb_filters=64, kernel_size=2, padding='causal', nb_stacks=1, dilations=None, activation='norm_relu', use_skip_connections=True, dropout_rate=0.0, return_sequences=True, name='tcn')

**Comment by philipperemy** [comment]
@qlemaire22 I made minor comments. We're almost good to go!

**Comment by qlemaire22** [comment]
The changes have been made! :)

**Comment by philipperemy** [comment]
Merged! Thanks a LOT!
